dnsapi/dns_wedos.sh

```diff
@@ -0,0 +1,366 @@
+      _err "After you export those variables, run the script again, the values will be saved for future"
+      return 1
+    fi
+
+    #build WEDOS_Authtoken
+    _debug "WEDOS Authtoken were not saved yet, building"
+    WEDOS_Authtoken=$(printf '%s' "${WEDOS_Wapipass}" | sha1sum | head -c 40)
+    _debug "WEDOS_Authtoken step 1, WAPI PASS sha1 sum: '${WEDOS_Authtoken}'"
+    WEDOS_Authtoken="${WEDOS_Username}${WEDOS_Authtoken}"
+    _debug "WEDOS_Authtoken step 2, username concat with token without hours: '${WEDOS_Authtoken}'"
+
+    #save details
+
+    _saveaccountconf_mutable WEDOS_Username "${WEDOS_Username}"
+    _saveaccountconf_mutable WEDOS_Wapipass "${WEDOS_Wapipass}"
+    _saveaccountconf_mutable WEDOS_Authtoken "${WEDOS_Authtoken}"
+  fi
+
+  if ! _get_root "${fulldomain}"; then
+    _err "WEDOS Account do not contain primary domain to fullfill add of ${fulldomain}!"
+    return 1
+  fi
+
+  _debug _sub_domain "${_sub_domain}"
+  _debug _domain "${_domain}"
+
+  if _wapi_find_row "${_domain}" "${_sub_domain}" "${txtvalue}"; then
+    _info "WEDOS WAPI: dns record found with id '${_row_id}'"
+
+    if _wapi_delete_row "${_domain}" "${_row_id}"; then
+      _info "WEDOS WAPI: dns row were deleted and changes commited!"
+      return 0
+    fi
+  fi
+
+  _err "Requested dns row were not found or was imposible to delete it, do it manually"
+  _err "Delete: ${fulldomain}"
+  _err "Value: ${txtvalue}"
+  return 1
+}
+
+####################  Private functions below ##################################
+
+# Function _wapi_post(), only takes data, prepares auth token and provide result
+# $1 - WAPI command string, like 'dns-domains-list'
+# $2 - WAPI data for given command, is not required
+# returns WAPI response if request were successfully delivered to WAPI endpoint
+_wapi_post() {
+  command=$1
+  data=$2
+
+  _debug "Command : ${command}"
+  _debug "Data : ${data}"
+
+  if [ -z "${command}" ]; then
+    _err "No command were provided, implamantation error!"
+    return 1
+  fi
+
+  # Prepare authentification token
+  hour=$(date +%H)
+  token=$(printf '%s' "${WEDOS_Authtoken}${hour}" | sha1sum | head -c 40)
+  _debug "Authentification token is '${token}'"
+
+  # Build xml request
+
+  request="request=<?xml version=\"1.0\" encoding=\"UTF-8\"?>\
+<request>\
+  <user>${WEDOS_Username}</user>\
+  <auth>${token}</auth>\
+  <command>${command}</command>"
+
+  if [ -z "${data}" ]; then
+    echo "" 1>/dev/null
+  else
+    request="${request}${data}"
+  fi
+
+  if [ -z "$TESTING_STAGE" ]; then
+    echo "" 1>/dev/null
+  else
+    request="${request}\
+  <test>1</test>"
+  fi
+
+  request="${request}\
+</request>"
+
+  _debug "Request to WAPI is: ${request}"
+
+  if ! response="$(_post "${request}" "$WEDOS_WAPI_ENDPOINT")"; then
+    _err "Error contacting WEDOS WAPI with command ${command}"
+    return 1
+  fi
+
+  _debug "Response : ${response}"
+  echo "${response}" | grep "<code>1000</code>" 1>/dev/null 2>/dev/null
```

Comment:
use `_contains()` function.

Reply:
Implemented, sorry I miss the wiki guide to implement it.